android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/UpcomingServicesAdapter.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.anc.ui.anccare.details
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import org.smartregister.fhircore.anc.data.anc.model.UpcomingServiceItem
+import org.smartregister.fhircore.anc.databinding.ItemServicesBinding
+
+/** Subclass of [ListAdapter] used to display careplan for the ANC client */
+class UpcomingServicesAdapter :
+  ListAdapter<UpcomingServiceItem, UpcomingServicesAdapter.PatientUpcomingServiceViewHolder>(
+    ImmunizationItemDiffCallback
```

Comment:
Is this only for Immunizations or does it work more generically?

Reply:
This is currently being used upcoming service section of anc patient but it can be used for any patient's upcoming services.

Reply:
Same here, this code needs to be refactored as every adapter has it's own DiffCallback.